gravity.sh

```diff
@@ -1,482 +1,661 @@
+# Determine if superseded pihole.conf exists
+if [[ -r "${piholeDir}/pihole.conf" ]]; then
+  echo -e "  ${COL_LIGHT_RED}Ignoring overrides specified within pihole.conf! ${COL_NC}"
 fi
 
-###########################
-# Collapse - begin formation of pihole
-gravity_collapse() {
-
-  #New Logic:
-  # Does /etc/pihole/adlists.list exist? If so leave it alone
-  #                                      If not, cp /etc/.pihole/adlists.default /etc/pihole/adlists.list
-  # Read from adlists.list
+# Determine if DNS resolution is available before proceeding
+gravity_DNSLookup() {
+  local lookupDomain="pi.hole" plural=""
 
-  # The following two blocks will sort out any missing adlists in the /etc/pihole directory, and remove legacy adlists.default
-  if [[ -f "${adListDefault}" ]] && [[ -f "${adListFile}" ]]; then
-    rm "${adListDefault}"
+  # Determine if $localList does not exist
+  if [[ ! -e "${localList}" ]]; then
+    lookupDomain="raw.githubusercontent.com"
   fi
 
-  if [ ! -f "${adListFile}" ]; then
-    cp "${adListRepoDefault}" "${adListFile}"
-  fi
-
-  echo -e "  ${INFO} Neutrino emissions detected..."
-  echo ""
-  local str="Pulling source lists into range"
-  echo -ne "  ${INFO} ${str}..."
-
-  sources=()
-  while IFS= read -r line || [[ -n "$line" ]]; do
-    # Do not read commented out or blank lines
-    if [[ ${line} = \#* ]] || [[ ! ${line} ]]; then
-      echo "" > /dev/null
-    else
-      sources+=(${line})
+  # Determine if $lookupDomain is resolvable
+  if timeout 1 getent hosts "${lookupDomain}" &> /dev/null; then
+    # Print confirmation of resolvability if it had previously failed
+    if [[ -n "${secs:-}" ]]; then
+      echo -e "${OVER}  ${TICK} DNS resolution is now available\\n"
     fi
-  done < ${adListFile}
-
-  echo -e "${OVER}  ${TICK} ${str}"
-}
+    return 0
+  elif [[ -n "${secs:-}" ]]; then
+    echo -e "${OVER}  ${CROSS} DNS resolution is not available"
+    exit 1
+  fi
 
-# patternCheck - check to see if curl downloaded any new files.
-gravity_patternCheck() {
-  patternBuffer=$1
-  success=$2
-  error=$3
-  if [[ "${success}" = true ]]; then
-    # Check if download was successful but list has not been modified
-    if [[ "${error}" == "304" ]]; then
-      echo -e "  ${TICK} No changes detected, transport skipped!"
-    # Check if the patternbuffer is a non-zero length file
-    elif [[ -s "${patternBuffer}" ]]; then
-      # Some blocklists are copyright, they need to be downloaded and stored
-      # as is. They can be processed for content after they have been saved.
-      mv "${patternBuffer}" "${saveLocation}"
-      echo -e "  ${TICK} List updated, transport successful!"
-    else
-      # Empty file -> use previously downloaded list
-      echo -e "  ${INFO} Received empty file, ${COL_LIGHT_GREEN}using cached one${COL_NC} (list not updated!)"
-    fi
+  # Determine error output message
+  if pidof dnsmasq &> /dev/null; then
```

Comment:
@WaLLy3K correct me if I'm wrong, but this is saying if `dnsmasq` has a PID, then the message _DNS resolution is currently unavailable_ is displayed to the user.  But if `dnsmasq` _is_ running, then resolution would be possible.

Reply:
https://github.com/pi-hole/pi-hole/blob/47099e2855884feb782479172803cf3e4baf4297/gravity.sh#L79-L85

If resolution is available, the function is exited using `return 0`, otherwise the function continues as if resolution is not available.